web/iot/worker_device_sim.js

```diff
@@ -65,12 +66,38 @@ function worker_node(){
       else if (args.data.action === 'finalizeProvisioning') {
         var subKey = args.data.params.sub
         var pubKey = args.data.params.pub
+        var url = args.data.params.url
         localPubNub = new PubNub({
           publishKey: pubKey,
           subscribeKey: subKey,
           uuid: id,
           listenToBrowserNetworkEvents: false //  Allows us to call the PubNub SDK from a web worker
-        })
+        });
+
+        // Create Simulator Metadata
+        await localPubNub.objects.setUUIDMetadata({
+          data: {
+            name: defaultDeviceName,
+            profileUrl: url
+          }
+        });
+
+        await localPubNub.objects.setChannelMembers({
```

Comment:
You also need to remove the channel members somewhere in your lifecycle, otherwise every time the user navigates to the page it will generate new members, will it not?

Reply:
Resolved